src/tx.rs

```diff
@@ -4,26 +4,25 @@
 //
 // Copyright (c) DUSK NETWORK. All rights reserved.
 
-use crate::{NodeClient, POSEIDON_DEPTH};
+use crate::NodeClient;
 
 use alloc::vec::Vec;
 use core::mem;
 
-use blake2b_simd::Params;
 use canonical::{Canon, Sink, Source};
-use dusk_abi::ContractId;
 use dusk_bytes::{
     DeserializableSlice, Error as BytesError, Serializable, Write,
 };
 use dusk_jubjub::{BlsScalar, JubJubAffine, JubJubExtended};
 use dusk_pki::{Ownable, SecretSpendKey};
 use dusk_plonk::prelude::{JubJubScalar, Proof};
-use dusk_poseidon::cipher::PoseidonCipher;
-use dusk_poseidon::sponge::hash;
 use dusk_poseidon::tree::PoseidonBranch;
 use dusk_schnorr::Proof as SchnorrSig;
 use phoenix_core::{Crossover, Fee, Note};
 use rand_core::{CryptoRng, RngCore};
+use rusk_abi::hash::Hasher;
+use rusk_abi::ContractId;
+use rusk_abi::POSEIDON_TREE_DEPTH;
```

Comment:
Nit: this two could be collapsed: `rusk_abi::{ContractId, POSEIDON_TREE_DEPTH}`.